### PR TITLE
Add TailwindCss Preset

### PR DESCRIPTION
**Pull request by simoebenhida**
User can confirm if he wants to add Tailwindcss or Not

### CODE REVIEW

**Comment by sixlive** [comment]
I'm down for this! I pushed some major updates and refactors to the Preset. If you're keen on updating this PR to match that I would be willing to merge.

File: src/stubs/webpack.mix.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+let mix = require('laravel-mix')
+require('laravel-mix-purgecss')
+
+mix.js('resources/js/app.js', 'public/js')
+  .postCss('resources/css/app.css', 'public/css')
+  .options({
+    postCss: [
+      require('postcss-import')(),
+      require('tailwindcss')(/* './path/to/tailwind.js' */),
```
Comment (sixlive) [inline src/stubs/webpack.mix.js]: We should fill this value in and assume that its in the `base_path()`

**Review by sixlive — DISMISSED** [review]
I'm thinking if we opt for the tailwind installation we should also just go ahead and run the the following as a task:

* `yarn install`
* `yarn tailwind init`
* `yarn dev`

**Review by sixlive — CHANGES_REQUESTED** [review]
I'm thinking if we opt for the tailwind installation we should also just go ahead and run the the following as a task:

* `yarn install`
* `yarn tailwind init`
* `yarn dev`

**Comment by simoebenhida** [comment]
Here u go I pulled the new changes and added a task for each command above. Let me know what you think 👍

File: src/Preset.php [hunk 2]
```diff
@@ -73,6 +73,22 @@ public function run()
             $this->runCommand('composer dumpautoload');
         });
 
+        if($this->command->confirm('Install Tailwindcss', true)) {
+            static::addTailwindcss($this->command);
```
Comment (sixlive) [inline src/Preset.php]:
```suggestion
           $this->command->task('Install Tailwindcss', function () {
               TailwindPreset::install();
           });
```

File: src/Preset.php [hunk 3]
```diff
@@ -73,6 +73,22 @@ public function run()
             $this->runCommand('composer dumpautoload');
         });
 
+        if($this->command->confirm('Install Tailwindcss', true)) {
```
Comment (sixlive) [inline src/Preset.php]:
```suggestion
        if($this->command->confirm('Install Tailwindcss?', true)) {
```

File: src/Preset.php [hunk 4]
```diff
@@ -73,6 +73,22 @@ public function run()
...
+            static::addTailwindcss($this->command);
+
+            $this->command->task('yarn install', function () {
+                $this->runCommand('yarn install');
+            });
+
+            $this->command->task('Setup Tailwindcss', function () {
+                $this->runCommand('yarn install');
```
Comment (sixlive) [inline src/Preset.php]:
```suggestion
                $this->runCommand('yarn tailwind init');
```

File: src/Preset.php [hunk 5]
```diff
@@ -203,6 +219,13 @@ private function runCommand($command)
         return exec(sprintf('%s 2>&1', $command));
     }
 
+    private function addTailwindcss()
```
Comment (sixlive) [inline src/Preset.php]: We can remove this method since we are adding the task for it above.

**Review by sixlive — CHANGES_REQUESTED** [review]
This is looking awesome. Just a few changes and we should be good.

File: src/Preset.php [hunk 6]
```diff
@@ -73,6 +73,24 @@ public function run()
             $this->runCommand('composer dumpautoload');
         });
 
+        if($this->command->confirm('Install Tailwindcss?', true)) {
+           $this->command->task('Install Tailwindcss', function () {
+               TailwindPreset::install();
+           });
+
+            $this->command->task('yarn install', function () {
```
Comment (sixlive) [inline src/Preset.php]:
```suggestion
            $this->command->task('Install node dependencies with Yarn', function () {
```

File: src/Preset.php [hunk 7]
```diff
@@ -73,6 +73,24 @@ public function run()
...
+                $this->runCommand('yarn install');
+            });
+
+            $this->command->task('Setup Tailwindcss', function () {
+                $this->runCommand('yarn tailwind init');
+            });
+
+            $this->command->task('yarn dev', function () {
```
Comment (sixlive) [inline src/Preset.php]:
```suggestion
            $this->command->task('Run node dev build with Yarn', function () {
```

**Review by sixlive — CHANGES_REQUESTED** [review]
Minor wording changes

**Comment by simoebenhida** [comment]
Awesome !! thanks for your reviews. We should that often 😄

**Review by sixlive — APPROVED** [review]

**Pull request merged**